python-packages/order_utils/src/zero_ex/abi_utils.py

```diff
@@ -0,0 +1,113 @@
+"""Ethereum ABI utilities.
+
+Builds on the eth-abi package, adding some convenience methods like those found
+in npmjs.com/package/ethereumjs-abi.  Ideally, all of this code should be
+pushed upstream into eth-abi.
+"""
+
+import re
+from typing import Any, List
+
+from mypy_extensions import TypedDict
+
+from eth_abi import encode_abi  # type: ignore
+from web3 import Web3  # type: ignore
+
+
+class MethodSignature(TypedDict, total=False):
+    """Object interface to an ABI method signature."""
+
+    method: str
+    args: List[str]
+
+
+def parse_signature(signature: str) -> MethodSignature:
+    """Parse a method signature into its constituent parts.
+
+    >>> parse_signature("ERC20Token(address)")
+    {'method': 'ERC20Token', 'args': ['address']}
+    """
+    if not isinstance(signature, str):
+        raise TypeError(
+            "signature should be str, not " + type(signature).__name__
+        )
+
+    matches = re.match(r"^(\w+)\((.+)\)$", signature)
+    if matches is None:
+        raise ValueError("Invalid method signature " + signature)
+    return {"method": matches[1], "args": matches[2].split(",")}
+
+
+def elementary_name(name: str) -> str:
+    """Convert from short to canonical names; barely implemented.
+
+    Modeled after ethereumjs-abi's ABI.elementaryName(), but only implemented
+    to support our particular use case and a few other simple ones.
+
+    >>> elementary_name("address")
+    'address'
+    >>> elementary_name("uint")
+    'uint256'
+    """
+    if not isinstance(name, str):
+        raise TypeError("name should be str, not " + type(name).__name__)
+
+    return {
+        "int": "int256",
+        "uint": "uint256",
+        "fixed": "fixed128x128",
+        "ufixed": "ufixed128x128",
+    }.get(name, name)
```

Comment:
Will this return an intelligible error message if it's not found? From the docs: 

```
Return the value for key if key is in the dictionary, else default. If default is not given, it defaults to None, so that this method never raises a KeyError.
```
https://docs.python.org/2/library/stdtypes.html#dict.get

I think we want to throw if returned `None`. 

General feedback: Let's prefer robust code with clear errors for library users over succinctness.

Reply:
I agree this looks weird, but I'm also not terribly familiar with all the pythonic idioms, so I'm not sure how many pythonistas would read it clearly.  There is no switch statement in python, and this was the alternative I liked best.  The second parameter to the `get()` method is the `default`, so here if the value of `name` isn't found in the dict, then `get()` will just return that value.  A naive approach might be to use `}[name]`, and then you'd have to worry about `KeyError`s, but [it seems that the whole purpose of `.get()` is for this specific case](https://stackoverflow.com/a/11041421/406289).

Reply:
Hm... I don't think falling back to the `name` passed in makes sense here. This way the dev will expect it to map and it won't but also doesn't throw, so they will only find out later on in their code.

Let's check if `get` returned `None`, and if so, throw loudly.

Reply:
This method is essentially a port of [`ethereumjs-util`s `elementaryName`](https://github.com/ethereumjs/ethereumjs-abi/blob/00ba8463a7f7a67fcad737ff9c2ebd95643427f7/lib/index.js#L9), which returns what was given if it doesn't know of a mapping.  And we depend on this behavior in our use case, where we pass in `"address"`, which doesn't have a mapping.  My understanding is that some types (like `address`) have the same representation in both `short name` and `canonical name` forms, so no mapping/translation needs to be done for those types.  And of course you could pass in types that area already in `canonical name` form, such as `uint256`, which would need no translation.  In short, I think it would be excessive to explicitly code for all of the types that should be supported, and that a "pass-through" functionality, at this relatively low level, is perfectly appropriate.  That's my opinion, and why I have it, but I'm open to having my mind changed.